torch/profiler/profiler.py

```diff
@@ -293,12 +296,26 @@ def step(self):
             self.step_rec_fn = prof.record_function("ProfilerStep#" + str(self.step_num))
             self.step_rec_fn.__enter__()
 
-    def export_chrome_trace(self, path: str):
+    def export_chrome_trace(self, path: str, use_gzip: bool = False):
         """
         Exports the collected trace in Chrome JSON format.
         """
         assert self.profiler
-        return self.profiler.export_chrome_trace(path)
+
+        if use_gzip and not path.endswith('.gz'):
+            path = path + '.gz'
+
+        if path.endswith('.gz'):
```

Comment:
why not just `if use_gzip` ?

Reply:
I add the ".gz" here to avoid user use .json to store the gzip contents. for example, one file is named <run_name>.json but it content is saved as gz. 

the tb_plugin actually use the extension ".gz" to check whether the file is json or gzip file

Reply:
I see, given all this complexity (we basically have two ways to signal gz) I think that we should better revert back to the original version that adds gz in the handler and export_chrome_trace accepts only the pathname - at least it's very straightforward